src/barplot.ts

```diff
@@ -69,41 +78,61 @@ export class BarPlot extends AVisInstance implements IVisInstance {
     return <Element>this.$node.node();
   }
 
+  transform(scale?: [number, number], rotate: number = 0) {
+    const bak = {
+      scale: this.options.scale || [1, 1],
```

Comment:
It's enough to do `array.slice()` ({the 0 is optional](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice)), because it only contains numbers.

`scale: this.options.scale.slice() || [1, 1],`

If you use this as multiple locations, change it everywhere.